decorator_factory.py

```diff
@@ -0,0 +1,21 @@
+def type_check(correct_type):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            if all(isinstance(arg, correct_type) for arg in args):
+                return func(*args, **kwargs)
+            else:
+                raise TypeError(f"All arguments must be of type {correct_type.__name__}")
+        return wrapper
+    return decorator
+
+
+@type_check(int)
+def times2(num):

```

Comment:
Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing